OpenRA.Mods.Common/Traits/Conditions/PausableConditionalTrait.cs

```diff
@@ -0,0 +1,81 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Primitives;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	/// <summary>Use as base class for *Info to subclass of PausableConditionalTrait. (See PausableConditionalTrait.)</summary>
+	public abstract class PausableConditionalTraitInfo : ConditionalTraitInfo
+	{
+		[ConsumedConditionReference]
+		[Desc("Boolean expression defining the condition to pause this trait.")]
+		public readonly BooleanExpression PauseOnCondition = null;
```

Comment:
`PauseOnCondition` is probably better, at least its consistent with `RequiresCondition` vs `RequireCondition`.